//----------------------------------------------------------------------------
// Copyright 2021, Ed Keenan, all rights reserved.
//----------------------------------------------------------------------------

syntax = "proto3";

import "vboData.proto";
import "textureData.proto";
import "animData.proto";

option optimize_for = LITE_RUNTIME;

message meshData_proto
{
	enum RENDER_MODE
	{
		MODE_POINTS			= 0;
		MODE_LINE			= 1;
		MODE_LINE_LOOP		= 2;
		MODE_LINE_STRIP     = 3;
		MODE_TRIANGLES      = 4;
		MODE_TRIANGLE_STRIP = 5;
		MODE_TRIANGLE_FAN   = 6;
	};

			 bytes			    pVersion			= 1;
	repeated bytes			    pName				= 2;
	repeated RENDER_MODE	    mode				= 3;
	repeated uint32             nodeNumber			= 4;
	repeated uint32			    triCount			= 5;
	repeated uint32			    vertCount			= 6;
	repeated vboData_proto      vbo_vert			= 7;
	repeated vboData_proto	    vbo_norm			= 8;
	repeated vboData_proto      vbo_uv				= 9;
	repeated vboData_proto      vbo_index			= 10;
	repeated vboData_proto      vbo_color			= 11;
	repeated textureData_proto  text_color			= 12;
	repeated animData_proto		anim_data			= 13;
	repeated uint32			    materialIndex		= 14;
			 uint32			    meshCount			= 15;
			 uint32			    texCount			= 16;
			 uint32			    nameCount			= 17;
			 uint32				animCount			= 18;
}													 
// -- End of File ---
